### PR TITLE
Apply dark window borders to NonClientIslandWindow using ThemeHelpers

### LINKED ISSUE #3425: Terminal has white borders in dark mode (Update ThemeUtils::SetWindowFrameDarkMode to use new DWM API)
From #3394:
> I'd be most comfortable merging this without the dark theme fix (which I know will cause there to be white borders around the entire window instead of just at the bottom) and then bringing in the dark theme changes when the other fix has flowed back down from Windows.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This solution was vetted by the DWM team.

Fixes #3425.

### CODE REVIEW

File: src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp [hunk 1]
```diff
@@ -485,6 +486,9 @@ void NonClientIslandWindow::_UpdateFrameMargins() const noexcept
         return _OnNcHitTest({ GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam) });
     case WM_PAINT:
         return _OnPaint();
+    case WM_ACTIVATE:
+        // If we do this every time we're activated, it should be close enough to correct.
+        TerminalTrySetDarkTheme(_window.get());
```
Comment (DHowett-MSFT) [inline src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: We don't care if this fails.

Comment (DHowett-MSFT) [reply src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp]: Not even enough to log it.

**Comment by zadjii-msft** [comment]
@msftbot merge this in 2 minutes

**Comment by DHowett-MSFT** [comment]
I made sure that this put the dll file in the x64 appx layout, but I'll be watching our prod builds to make sure.

**Pull request merged**